src/derivative_operators/derivative_operator.jl

```diff
@@ -45,10 +45,10 @@ end
 
 function negate!(arr::T) where T
     if size(arr,2) == 1
-        scale!(arr,-one(eltype(arr)))
+        scale!(arr,-one(eltype(arr[1]))) #fix right neumann bc, eltype(Vector{T}) doesnt work.
```

Comment:
Really? `eltype(Vector{T})` returns `T`.

Reply:
Sorry, what I meant what doesn't work is this:``eltype(arr)=eltype(Array{Array{T,1},1})=Array{T,1}`` and the ``one`` method fails.

Reply:
Got it.